src/app/units/states/tasks/inbox/directives/staff-task-list/staff-task-list.component.ts

```diff
@@ -389,7 +389,7 @@ export class StaffTaskListComponent implements OnInit, OnChanges {
     }
   }
 
-  togglePin(task) {
-    task.pinned ? task.unpinTask(task) : task.pinTask(task);
+  togglePin() {
```

Comment:
This should be passed the task to pin - it is not always the current selected task

Reply:
You're right @macite. @kumar-ish can you take a look what's on the filtered task object. If not might need to look at going through the task service or building out the Task object from the id in the filtered task

Reply:
Let me know if you need help

Reply:
Done -- sorry, I should have noticed this.